test-framework/core/src/main/java/org/keycloak/testframework/admin/AdminClientFactory.java

```diff
@@ -17,6 +24,21 @@ public class AdminClientFactory {
         delegateSupplier = () -> KeycloakBuilder.builder().serverUrl(serverUrl);
     }
 
+    AdminClientFactory(String serverUrl, KeyStore serverKeyStore) {
+        try {
+            SSLContext sslContext = SSLContextBuilder.create()
+                    .loadTrustMaterial(serverKeyStore, null)
+                    .build();
+
+            delegateSupplier = () ->
+                    KeycloakBuilder.builder()
+                            .serverUrl(serverUrl)
+                            .resteasyClient(Keycloak.getClientProvider().newRestEasyClient(null, sslContext, false));
+        } catch (NoSuchAlgorithmException | KeyStoreException | KeyManagementException e) {
+            throw new ManagedCertificatesException(e);
+        }
```

Comment:
You don't need this once you add `ManagedCertificates#getClientSSLContext` as that can then just catch the exceptions and throw the runtime `ManagedCertificatesException`